generators/server/templates/src/main/java/package/web/filter/ModifyServersOpenApiFilter.java.ejs

```diff
@@ -106,12 +115,66 @@ public class ModifyServersOpenApiFilter implements GlobalFilter, Ordered {
                 servers.add(serversToJson);
                 ((ObjectNode) jsonBody).set("servers", servers);
 
-                this.rewritedBody = jsonBody.toString();
-                originalResponse.getHeaders().setContentLength(this.rewritedBody.getBytes().length);
+                rewritedBody = jsonBody.toString();
+                return rewritedBodyToDataBuffer();
             } catch (JsonProcessingException e) {
                 e.printStackTrace();
             }
-            return bufferFactory.wrap(this.rewritedBody.getBytes());
+            return join;
+        }
+
+        private DataBuffer rewritedBodyToDataBuffer() {
+            if(isZippedResponse()) {
+                byte[] zippedBody = zipContent(rewritedBody);
+                originalResponse.getHeaders().setContentLength(zippedBody.length);
+                return bufferFactory.wrap(zippedBody);
+            } else {
+                originalResponse.getHeaders().setContentLength(rewritedBody.getBytes().length);
+                return bufferFactory.wrap(rewritedBody.getBytes());
+            }
+        }
+
+        private String contentToString(byte[] content) {
+            if(isZippedResponse()) {
+                byte[] unzippedContent = unzipContent(content);
+                return new String(unzippedContent, StandardCharsets.UTF_8);
+            } else {
+                return new String(content, StandardCharsets.UTF_8);
+            }
+        }
+
+        private boolean isZippedResponse() {
+            if (originalResponse.getHeaders().isEmpty() ||
```

Comment:
i think the logic here can be improved and be done directly with 1 return

Reply:
done